src/components/legal/legal.tsx

```diff
@@ -0,0 +1,110 @@
+import { INavigationProps } from '../../navigation/with-navigation-params';
+import { TC } from './terms-conditions/terms-conditions';
+import { PrivacyPolicy } from './privacy-policy/privacy-policy';
+import { IReduxState } from '../../redux/state';
+
+interface IExternalProps {
+    onAccept: () => void;
+}
+
+export interface IReduxProps {
+    tcVersion: number;
+    tcAcceptedVersion: number;
+    appSetTcVersion: typeof appSetTcVersion;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    return {
+        tcAcceptedVersion: state.app.tcAcceptedVersion
+    };
+};
+
+const mapDispatchToProps = {
+    appSetTcVersion
+};
+
+export const LegalComponent = (
+    props: INavigationProps &
+        IReduxProps &
+        IExternalProps &
+        IThemeProps<ReturnType<typeof stylesProvider>>
+) => {
+    const [showTC, setShowTC] = useState<boolean>(false);
+    const [showPrivacyPolicy, setShowPrivacyPolicy] = useState<boolean>(false);
+
+    if (showTC) {
+        return <TC showClose onClose={() => setShowTC(false)} />;
+    } else if (showPrivacyPolicy) {
+        return <PrivacyPolicy showClose onClose={() => setShowPrivacyPolicy(false)} />;
+    } else
+        return (
+            <View style={props.styles.container}>
+                <View style={props.styles.topContainer}>
+                    <Text style={props.styles.walletTc}>{translate('CreateWalletTc.body')}</Text>
+                    <Image
+                        resizeMode="contain"
+                        source={require('../../assets/images/png/document.png')}
+                        style={props.styles.docImage}
+                    />
+                </View>
+
+                <View style={props.styles.bottomContainer}>
+                    <TouchableOpacity
+                        style={props.styles.rowContainer}
+                        onPress={() => setShowTC(true)}
+                    >
+                        <Text style={props.styles.text}>{translate('App.labels.tc')}</Text>
+                        <Icon name="chevron-right" size={16} style={props.styles.icon} />
+                    </TouchableOpacity>
+
+                    <View style={props.styles.divider} />
+
+                    <TouchableOpacity
+                        style={props.styles.rowContainer}
+                        onPress={() => setShowPrivacyPolicy(true)}
+                    >
+                        <Text style={props.styles.text}>
+                            {translate('App.labels.privacyPolicy')}
+                        </Text>
+                        <Icon name="chevron-right" size={16} style={props.styles.icon} />
+                    </TouchableOpacity>
+
+                    <View style={props.styles.divider} />
+
+                    <Button
+                        style={props.styles.bottomButton}
+                        primary
+                        onPress={() => {
+                            props.appSetTcVersion(props.tcAcceptedVersion);
```

Comment:
i think this is not quite correct, `props.tcAcceptedVersion` are the t&c that were already accepted